discovery/consul/consul.go

```diff
@@ -106,7 +107,13 @@ func NewDiscovery(conf *config.ConsulSDConfig, logger log.Logger) (*Discovery, e
 	if err != nil {
 		return nil, err
 	}
-	transport := &http.Transport{TLSClientConfig: tls}
+	transport := &http.Transport{
+		TLSClientConfig: tls,
+		DialContext: conntrack.NewDialContextFunc(
+			conntrack.DialWithTracing(),
+			conntrack.DialWithName("consul"),
```

Comment:
consul_sd would be better, as if this ends up being used in another application with other consul use that'll be clearer.

Reply:
done, also for marathon and triton discoveries.